app/integration_action_test.go

```diff
@@ -469,6 +469,7 @@ func TestSubmitInteractiveDialog(t *testing.T) {
 			plugin.ClientMain(&MyPlugin{})
 		}
 		`, `{"id": "myplugin", "backend": {"executable": "backend.exe"}}`, "myplugin", th.App)
+	tearDown()
```

Comment:
Yes, this one looks a bit weird, it feels like we need to put a defer on it. I totally agree with you, doing the cleanups after the whole Test func is completed makes the most sense.

But I haven't deferred them because I didn't wan't to change the previous behavior where _temp_ dirs gets deleted after `setupPluginApiTest()` returns.

Tho, I tried once deferring it all but I remember that tests did fail and I thought there might be some logic depends on non-deferred use -actually might even be a bug- so I reverted back.

Reply:
Agreed that there might be some logic that depends on the weird behaviour here. If we end up needing to just call `tearDown()` right away, is there any reason to return the function at all? Why not just continue to teardown in the function itself?

Reply:
The new test I have added depends on deferred _tearDown()_.

I agree that the whole situation is ambiguous.

If you would like that, I can revert my changes on [setupMultiPluginApiTest()](https://github.com/mattermost/mattermost-server/pull/13041/files#diff-8a13870e4ccd1878b9c6bae995211f3fR31), don't use it, and just duplicate some code from that for the new test. 

After we figure out the whole situation, we can refactor the test again to get rid of duplicated _setup_ code.

Reply:
Ah, that makes sense. I'd advocate for duplicating the code for new and just revisiting this setup code as it's always been a bit obtuse.

Reply:
done by 5e71708114c180a966cee941ed54e39bc13e9c23